astropy/coordinates/matching.py

```diff
@@ -192,7 +226,48 @@ def match_coordinates_sky(
         # the old backwards-compatible name
         catalogcoord.cache["kdtree"] = newcat_u.cache["kdtree"]
 
-    return idx, sep2d, sep3d
+    return CoordinateMatchResult(idx, sep2d, sep3d)
+
+
+class CoordinateSearchResult(NamedTuple):
+    """Results of searching close pairs between two sets of sources.
+
+    Parameters
+    ----------
+    indices_to_first_set : int array
+        Indices of the elements of the found pairs in the first set of
+        sources.
+    indices_to_second_set : int array
+        Indices of the elements of the found pairs in the second set of
+        sources.
+    angular_separation : `~astropy.coordinates.Angle`
+        The angular separations between the paired sources.
+    physical_separation : `~astropy.units.Quantity`
+        The physical separations between the paired sources. If either
+        of the source sets lack distances then the physical separations
+        are computed assuming all coordinates are points on the unit
+        sphere.
+
+    See Also
+    --------
+    astropy.coordinates.search_around_3d
+    astropy.coordinates.search_around_sky
+    SkyCoord.search_around_3d
+    SkyCoord.search_around_sky
+    """
+
+    indices_to_first_set: NDArray[np.int64]
```

Comment:
The actual array is created with `dtype=int`, which can be 32 bit. I'd adjust it to match here (same in other `NamedTuple`).